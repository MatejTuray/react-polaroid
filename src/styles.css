/* add css styles here (optional) */
.Polaroid_container {
  perspective: 600px;
  position: relative;
}
.Polaroid_card {
  width: 100%;
  height: 100%;
  position: relative;
  transition: transform 1s;
  transform-style: preserve-3d;
  cursor: pointer;
}
.Polaroid_container_front {
  display: flex;
  position: absolute;
  backface-visibility: hidden;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);

  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}
.Polaroid_side_flip_card {
  transform-origin: center right;
}
.Polaroid_side_flip {
  transform: translateX(-100%) rotateY(-180deg);
}
.Polaroid_default_flip {
  transform: rotateY(180deg);
}
.Polaroid_container_back {
  display: flex;

  position: absolute;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.Polaroid_container_front:hover {
  box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
}
.Polaroid_container_back:hover {
  box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
}
.Polaroid_text {
  margin-bottom: 1rem;
  margin-left: 1rem;
}

.Polaroid_image {
  margin: 1rem;
  margin-top: 2rem;
}
