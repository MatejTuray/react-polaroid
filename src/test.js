import Polaroid from "./index.js";
import styles from "./styles.css";
import React from "react";
import renderer from "react-test-renderer";

const defaultProps = {
  height: 400,
  width: 220,
  imgSrc: "placecat.jpg",
  frontText: "Polaroid kitty - front",
  rotation: 0,
  backText: "Polaroid kitty - back",
  flip: false,
  disabled: false,
  type: "default",
  cardColor: "white"
};

describe("Correctly rendering Polaroid component - snapshot", () => {
  it("should render Polaroid correctly", () => {
    const PolaroidTest = renderer
      .create(
        <Polaroid
          frontText="Front text test"
          backText="Back text test"
          disabled={false}
          height={500}
          width={250}
          type={"side"}
          cardColor={"blue"}
        />
      )
      .toJSON();
    expect(PolaroidTest).toMatchSnapshot();
  });
});

describe("Testing props", () => {
  it("default props should be correct", () => {
    expect(JSON.stringify(Polaroid.defaultProps)).toBe(
      JSON.stringify(defaultProps)
    );
  });
  it("changing backText should be correct", () => {
    const PolaroidTest = renderer
      .create(
        <Polaroid
          frontText="Front text test"
          backText="Back text test"
          disabled={false}
          height={500}
          width={250}
          type={"side"}
          cardColor={"blue"}
        />
      )
      .toTree();

    expect(PolaroidTest.props.backText).toBe("Back text test");
  });
  it("changing frontText should be correct", () => {
    const PolaroidTest = renderer
      .create(
        <Polaroid
          frontText="Front text test"
          backText="Back text test"
          disabled={false}
          height={500}
          width={250}
          type={"side"}
          cardColor={"blue"}
        />
      )
      .toTree();

    expect(PolaroidTest.props.frontText).toBe("Front text test");
  });
  it("changing flipping types should be correct", () => {
    const PolaroidTest = renderer
      .create(
        <Polaroid
          frontText="Front text test"
          backText="Back text test"
          disabled={false}
          height={500}
          width={250}
          type={"side"}
          cardColor={"blue"}
        />
      )
      .toTree();

    expect(PolaroidTest.props.type).toBe("side");
  });
  it("changing card color should work", () => {
    const PolaroidTest = renderer
      .create(
        <Polaroid
          frontText="Front text test"
          backText="Back text test"
          disabled={false}
          height={500}
          width={250}
          type={"side"}
          cardColor={"#ff5f89"}
        />
      )
      .toTree();

    expect(PolaroidTest.props.cardColor).toBe("#ff5f89");
  });
  it("changing height should work", () => {
    const PolaroidTest = renderer
      .create(
        <Polaroid
          frontText="Front text test"
          backText="Back text test"
          disabled={false}
          height={500}
          width={250}
          type={"side"}
          cardColor={"#ff5f89"}
        />
      )
      .toTree();

    expect(PolaroidTest.props.height).toBe(500);
  });
  it("changing width should work", () => {
    const PolaroidTest = renderer
      .create(
        <Polaroid
          frontText="Front text test"
          backText="Back text test"
          disabled={false}
          height={500}
          width={250}
          type={"side"}
          cardColor={"#ff5f89"}
        />
      )
      .toTree();

    expect(PolaroidTest.props.width).toBe(250);
  });
});
