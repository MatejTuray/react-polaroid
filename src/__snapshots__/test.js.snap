// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly rendering Polaroid component - snapshot should render Polaroid correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "height": 500,
      "maxHeight": 500,
      "maxWidth": 250,
      "transform": "rotate(0deg)",
      "width": 250,
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "backgroundColor": "blue",
        "transform": "rotateY(0deg)",
        "transformOrigin": "center right",
      }
    }
  >
    <div
      className={undefined}
      style={Object {}}
    >
      <p
        className={undefined}
        style={
          Object {
            "transform": "rotateY(180deg)",
          }
        }
      >
        Back text test
      </p>
    </div>
    <div
      className={undefined}
      style={Object {}}
    >
      <img
        className={undefined}
        src="placecat.jpg"
      />
      <p
        className={undefined}
      >
        Front text test
      </p>
    </div>
  </div>
</div>
`;
